src/main/java/com/diconiumwvv/storesservice/stores/dtos/StoreDraftDTOToStoreDraftConverter.java

```diff
@@ -0,0 +1,39 @@
+package com.diconiumwvv.storesservice.stores.dtos;
+
+import io.sphere.sdk.channels.ChannelDraft;
+import io.sphere.sdk.channels.ChannelDraftBuilder;
+import io.sphere.sdk.types.CustomFieldsDraftBuilder;
+import org.springframework.core.convert.converter.Converter;
+
+import java.util.HashMap;
+import java.util.Optional;
+
+public class StoreDraftDTOToStoreDraftConverter implements Converter<StoreDraftDTO, ChannelDraft> {
+
+    @Override
+    public ChannelDraft convert(StoreDraftDTO storeDraftDTO) {
+        ChannelDraftBuilder draftBuilder = ChannelDraftBuilder.of(storeDraftDTO.getKey());
+
+        Optional.ofNullable(storeDraftDTO.getName()).ifPresent(draftBuilder::name);
+        Optional.ofNullable(storeDraftDTO.getDescription()).ifPresent(draftBuilder::description);
+        Optional.ofNullable(storeDraftDTO.getAddress()).ifPresent(draftBuilder::address);
+        Optional.ofNullable(storeDraftDTO.getGeoLocation()).ifPresent(draftBuilder::geoLocation);
+
+        draftBuilder.custom(
+                CustomFieldsDraftBuilder.ofTypeKey("channel-neighborhoods")
+                        .addObjects(new HashMap<String, Object>() {{
+                            Optional.ofNullable(storeDraftDTO.getNeighborhood()).ifPresent(o -> put("neighborhood", o));
```

Comment:
Instead of doing all the null checks here you could apply some validations on the DTO. 
And apply the @Valid annotation to the controller. 
Then the call will be rejected by spring boot in case some data is not valid. 
https://www.baeldung.com/spring-boot-bean-validation

Reply:
The problem is, that the parameters are optional, but the commercetools builder accepts null values and then fails on them when posting the request to the API. I found a somewhat more readable version of the null checks though.

I will talk through authentication with Thomas

Reply:
Ok, I think then it makes sense :D